src/ip/Address.cc

```diff
@@ -404,7 +404,11 @@ Ip::Address::lookupHostIP(const char *s, bool nodns)
         }
         if (maybeIpv4 != nullptr)
             res = maybeIpv4;
-        // else IPv6-only host, let the caller deal with first-IP anyway.
```

Comment:
Please update this PR branch to merge recent master/v7 changes (including commit 51c518d and especially commit 97bbba6). This PR title/description and code should be adjusted to shift its focus from fixing bug 5154 to other valuable improvements that this PR already implements. For example, AFAICT, PR changes in this function should be undone after that update because they go against commit 97bbba6 approach (and are no longer necessary to prevent bug 5154 assertions).